src/games/takesix/game.ts

```diff
@@ -0,0 +1,219 @@
+            ...G.decks[id], // Move current deck into penalty cards in player's deck
+          ],
+        },
+      }),
+      decks: Object.values({
+        ...G.decks,
+        [id]: [card], // Set selected card as current in selected deck
+      }),
+    };
+  } else {
+    const diffs: number[] = G.decks.map(
+      (deck: Card[]) => card.number - deck[deck.length - 1].number,
+    );
+
+    let min = Number.MAX_SAFE_INTEGER;
+    let minIndex = 0;
+    diffs.forEach((diff, index) => {
+      if (diff > 0 && diff < min) {
+        min = diff;
+        minIndex = index;
+      }
+    });
+
+    if (minIndex !== id) {
+      return INVALID_MOVE;
+    }
+
+    // If card is #6 move all cards from deck to player's hand
+    if (G.decks[id].length === 5) {
+      return {
+        ...G,
+        players: Object.values({
+          ...G.players,
+          [ctx.playerID]: {
+            ...G.players[ctx.playerID as any],
+            penaltyCards: [
+              ...G.players[ctx.playerID as any].penaltyCards,
+              ...G.decks[id],
+            ],
+          },
+        }),
+        decks: Object.values({
+          ...G.decks,
+          [id]: [card],
+        }),
+      };
+    }
+
+    return {
+      ...G,
+      decks: Object.values({
+        ...G.decks,
+        [id]: [...G.decks[id], card],
+      }),
+    };
+  }
+}
+
+export const TakeSixGame = Game({
+  name: 'takesix',
+  flow: {
+    endTurn: false,
+    endPhase: false,
+    endGame: false,
+    startingPhase: 'CARD_SELECT', // Start by selecting cards
+    phases: {
+      // Everyone needs to select card
+      CARD_SELECT: {
+        allowedMoves: ['selectCard'],
+        turnOrder: TurnOrder.ANY_ONCE,
+        next: 'DECK_SELECT',
+        // Determine player order
+        onPhaseEnd: (G: IG) => {
+          const selectedCards = G.players.map(
+            (player: Player) => player.selectedCard,
+          );
+          selectedCards.sort((a: Card, b: Card) => a.number - b.number);
+          return {
+            ...G,
+            cardOrder: selectedCards
+              .map((card: Card) => card.owner)
+              .map((x: number) => x.toString()),
+          };
+        },
+      },
+      // Select deck
+      DECK_SELECT: {
+        allowedMoves: ['selectDeck'],
+        next: 'CARD_SELECT',
+        // Implement CUSTOM_FROM_ONCE
+        turnOrder: {
+          playOrder: (G: IG) => G.cardOrder,
+          first: () => 0,
+          next: (_, ctx) => {
+            if (ctx.playOrderPos < ctx.playOrder.length - 1) {
+              return ctx.playOrderPos + 1;
+            }
+          },
+        },
+        onMove: (_, ctx) => {
+          ctx.events.endTurn();
+        },
+        onPhaseEnd: (G: IG) => {
+          if (G.players[0].cards.length === 0) {
+            G.end = true;
+          }
+        },
+      },
+    },
+    endGameIf: (G: IG) => {
+      if (G.end === true) {
+        const scoreboard = G.players.map((player, i) => ({
```

Comment:
you probably going to need to show the scoreboard in the UI too, so might be useful this to live in its own function